Load test failing on dedicated runner

### Component(s)

_No response_

### Describe the issue you're reporting

The following failure occurs when running the load test on dedicated runners:

```
2023/10/04 17:53:04 CPU consumption is 78.3%, max expected is 60%
    test_case.go:292: CPU consumption is 78.3%, max expected is 60%
    scenarios.go:304: 
        	Error Trace:	/home/ghrunner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/testbed/tests/scenarios.go:304
        	            				/home/ghrunner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/testbed/tests/trace_test.go:191
        	Error:      	"50" is not less than "0"
        	Test:       	TestTraceNoBackend10kSPS/MemoryLimit
2023/10/04 17:53:04 Stopped generator. Sent:     29090 items
--- FAIL: TestTraceNoBackend10kSPS (18.48s)
    --- PASS: TestTraceNoBackend10kSPS/NoMemoryLimit (15.28s)
    --- FAIL: TestTraceNoBackend10kSPS/MemoryLimit (3.20s)
```

https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/6408839554/job/17401287975

--- a/testbed/testbed/child_process_collector.go
+++ b/testbed/testbed/child_process_collector.go
@@ -204,6 +204,9 @@ func (cp *childProcessCollector) Start(params StartParams) error {
 	}
 	// #nosec
 	cp.cmd = exec.Command(exePath, args...)
+	cp.cmd.Env = append(os.Environ(),
+		"GOMAXPROCS=2",
+	)
 
 	// Capture standard output and standard error.
 	cp.cmd.Stdout = logFile

